pkg/k8s/init.go

```diff
@@ -31,6 +31,9 @@ func Init() error {
 	if err := createDefaultClient(); err != nil {
 		return fmt.Errorf("unable to create k8s client: %s", err)
 	}
+	if _, err := GetServerVersion(); err != nil {
```

Comment:
Can we add an error if k8s version is less than the minimal version supported by cilium? (`>1.6`)

Reply:
That is a good idea.  I'll add it.

Reply:
Done. I opted to bound the range on both ends: `> 1.6, < 1.11`. I'm not sure that this is a good idea, since it means we would break in the future. On the other hand, it explicitly enforces our supported ranges. What do you think?